retrieval/discovery/kubernetes/pod.go

```diff
@@ -74,7 +74,10 @@ func (p *Pod) Run(ctx context.Context, ch chan<- []*config.TargetGroup) {
 			send(p.buildPod(o.(*apiv1.Pod)))
 		},
 		DeleteFunc: func(o interface{}) {
-			send(&config.TargetGroup{Source: podSource(o.(*apiv1.Pod))})
+			pod, ok := o.(*apiv1.Pod)
```

Comment:
Yep, [sounds like](https://github.com/kubernetes/client-go/blob/843f7c4f28b1f647f664f883697107d5c02c5acc/1.5/tools/cache/delta_fifo.go#L631-L634) the type cast should be safe on the inner Obj. Will update.